File: abc393a/abc393a.ml
open Printf
open Scanf

let solve s1 s2 =
  match (s1, s2) with
    "fine", "fine" -> 4
  | "fine", "sick" -> 3
  | "sick", "fine" -> 2
  | _ -> 1

let () =
  scanf "%s %s " solve |> printf "%d\n"

File: abc393a/abc393a.hs
{-# LANGUAGE OverloadedStrings #-}

main :: IO ()
main = solve <$> words <$> getLine >>= print

solve :: [String] -> Int
solve ["fine", "fine"] = 4
solve ["fine", "sick"] = 3
solve ["sick", "fine"] = 2
solve _ = 1
